File: 009.PalindromeNumber/wyrj/PalindromeNumber.js
/* 784ms */
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if (0 > x) {
        return false;
    }
    var arr = [];
    while(0 < x) {
        arr.push(x%10);
        x = Math.floor(x/10);
    }
    var start = 0, end = arr.length - 1;
    while(start < end) {
        if (arr[start] !== arr[end]) {
            return false;
        }
        start++;
        end--;
    }
    return true;
};

File: 009.PalindromeNumber/wyrj/PalindromeNumber.c
/* 56ms */
bool isPalindrome(int x) {
    if (0 > x) {
        return false;
    }
    int arr[100], len = 0;
    while(0 < x) {
        arr[len++] = x % 10;
        x = x / 10;
    }
    int start = 0, end = len - 1;
    while(start < end) {
        if (arr[start] != arr[end]) {
            return false;
        }
        start++;
        end--;
    }
    return true;
}